crates/driver/src/tests/setup/mod.rs

```diff
@@ -1372,13 +1383,19 @@ impl QuoteOk<'_> {
     /// Check that the quote returns the expected amount of tokens. This is
     /// based on the state of the blockchain and the test setup.
     pub fn amount(self) -> Self {
-        assert_eq!(self.trades.len(), 1);
-        let quoted_order = match &self.trades[0] {
-            Trade::Fulfillment(fulfillment) => &fulfillment.quoted_order,
-            Trade::Jit(jit) => &jit.quoted_order,
-        };
+        let trades = self
+            .trades
+            .iter()
+            .filter(|trade| matches!(trade, Trade::Fulfillment(_)))
+            .collect::<Vec<_>>();
+        assert_eq!(trades.len(), 1);
+
         let result: serde_json::Value = serde_json::from_str(&self.body).unwrap();
         let amount = result.get("amount").unwrap().as_str().unwrap().to_owned();
+        let quoted_order = match trades[0] {
```

Comment:
Looks like this match could be avoided by using `.filter_map` a few lines above to already get the `quoted_order` out.

Reply:
Thanks, completely forgot to clean up the code after the test started to pass finally.